cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -365,4 +377,41 @@ private void emitApplicationCountMetric() {
       LOG.warn("Failed to get application count", e);
     }
   }
+
+  private void emitPluginCountMetric() {
+    emitPluginCountMetric(null);
+  }
+
+  private void emitPluginCountMetric(Set<ImmutablePair<String, String>> deletedPlugins) {
+    try {
+      List<ImmutablePair<String, String>> remainingPlugins = appMetadataStore.getAllApplications().stream()
+        .map(ApplicationMeta::getSpec)
+        .map(ApplicationSpecification::getPlugins)
+        .flatMap(e -> e.values().stream())
+        .map(Plugin::getPluginClass)
+        .map(e -> ImmutablePair.of(e.getName(), e.getType()))
+        .collect(Collectors.toList());
+
+      Map<ImmutablePair<String, String>, Long> pluginCounts = remainingPlugins.stream()
+        .collect(Collectors.groupingByConcurrent(Function.identity(), Collectors.counting()));
+      for (Map.Entry<ImmutablePair<String, String>, Long> entry : pluginCounts.entrySet()) {
+        Map<String, String> tags = ImmutableMap.of(Constants.Metrics.Tag.PLUGIN_NAME, entry.getKey().getFirst(),
+                                                   Constants.Metrics.Tag.PLUGIN_TYPE, entry.getKey().getSecond());
+        metricsCollectionService.getContext(tags).gauge(Constants.Metrics.Program.PLUGIN_COUNT, entry.getValue());
```

Comment:
It seems like this is not tied to a pipeline right? So this will emit total plugin count across all pipelines in a given instance correct? I was under the impression we wanted to break it down by pipeline as well, is that not the case?